src/Features/Core/Portable/Diagnostics/CodeAnalysisDiagnosticAnalyzerService.cs

```diff
@@ -88,30 +89,14 @@ public void Clear()
         public bool HasProjectBeenAnalyzed(ProjectId projectId)
             => _analyzedProjectToDiagnostics.ContainsKey(projectId);
 
-        public async Task RunAnalysisAsync(Solution solution, ProjectId? projectId, Action<Project> onAfterProjectAnalyzed, CancellationToken cancellationToken)
+        public async ValueTask RunAnalysisAsync(Project project, CancellationToken cancellationToken)
         {
-            Contract.ThrowIfFalse(solution.Workspace == _workspace);
+            cancellationToken.ThrowIfCancellationRequested();
 
-            if (projectId != null)
```

Comment:
this logic went away.  this service *always* operates at a particular project level.  One caller exists that wants to operate at the Solution level.  So now *it* owns the Parallel.ForEachAsync for doing that.